spec/lib/stripe/credit_card_cloner_spec.rb

```diff
@@ -5,62 +5,62 @@
 
 module Stripe
   describe CreditCardCloner do
-    describe "#find_or_clone" do
-      include StripeStubs
-
-      let(:credit_card) { create(:credit_card, user: create(:user)) }
-      let(:stripe_account_id) { "abc123" }
+    let!(:user) { create(:user, email: "jumping.jane@example.com") }
+    let!(:enterprise) { create(:enterprise) }
 
-      let(:cloner) { Stripe::CreditCardCloner.new(credit_card, stripe_account_id) }
+    let(:secret) { ENV.fetch('STRIPE_SECRET_TEST_API_KEY', nil) }
 
-      let(:customer_id) { "cus_A123" }
-      let(:payment_method_id) { "pm_1234" }
-      let(:new_customer_id) { "cus_A456" }
-      let(:new_payment_method_id) { "pm_456" }
-      let(:payment_method_response_mock) { { status: 200, body: payment_method_response_body } }
+    describe "#find_or_clone", :vcr, :stripe_version do
+      include StripeStubs
 
-      let(:payment_method_response_body) {
-        JSON.generate(id: new_payment_method_id)
-      }
+      before { Stripe.api_key = secret }
 
-      before do
-        Stripe.api_key = "sk_test_12345"
+      let!(:customer_id) { ENV.fetch('STRIPE_CUSTOMER', nil) }
 
-        stub_customers_post_request email: credit_card.user.email,
-                                    response: { customer_id: new_customer_id },
-                                    stripe_account_header: true
+      let!(:stripe_account_id) { ENV.fetch('STRIPE_ACCOUNT', nil) }
 
-        stub_retrieve_payment_method_request(payment_method_id)
-        stub_list_customers_request(email: credit_card.user.email, response: {})
-        stub_get_customer_payment_methods_request(customer: "cus_A456", response: {})
-        stub_add_metadata_request(payment_method: "pm_456", response: {})
+      let!(:stripe_account) {
+        create(:stripe_account, enterprise:, stripe_user_id: stripe_account_id)
+      }
 
-        stub_request(:post,
-                     "https://api.stripe.com/v1/payment_methods/#{new_payment_method_id}/attach")
-          .with(body: { customer: new_customer_id },
-                headers: { 'Stripe-Account' => stripe_account_id })
-          .to_return(payment_method_response_mock)
+      let(:credit_card) { create(:credit_card, gateway_payment_profile_id: pm_card.id, user:) }
+
+      let(:pm_card) {
+        Stripe::PaymentMethod.create(
+          {
+            type: 'card',
+            card: {
+              number: '4242424242424242',
+              exp_month: 8,
+              exp_year: 2026,
```

Comment:
We'll need to update that in a couple of years.

Reply:
Ahh, yes I forgot the usual bit of setting a future date... I'll keep this in mind and change it in the upcoming PR.